indico/modules/events/registration/controllers/api/checkin.py

```diff
@@ -0,0 +1,109 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2023 CERN
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see the
+# LICENSE file for more details.
+
+from flask import request
+from sqlalchemy.orm import selectinload, undefer
+from webargs import fields
+
+from indico.core import signals
+from indico.modules.events.management.controllers.base import RHManageEventBase
+from indico.modules.events.payment.util import toggle_registration_payment
+from indico.modules.events.registration.models.forms import RegistrationForm
+from indico.modules.events.registration.models.registrations import Registration
+from indico.modules.events.registration.schemas import (CheckinEventSchema, CheckinRegFormSchema,
+                                                        CheckinRegistrationSchema)
+from indico.web.args import use_kwargs
+from indico.web.rh import cors, json_errors, oauth_scope
+
+
+@json_errors
+@cors
+@oauth_scope('registrants')
+class RHCheckinAPIBase(RHManageEventBase):
+    """Base class for the Check-in API."""
+
+    PERMISSION = 'registration'
+
+
+class RHCheckinAPIEventDetails(RHCheckinAPIBase):
```

Comment:
Maybe just `RHCheckinAPIEvent` and `RHCheckinAPIRegForm` would be sufficient? `...Details` is only used for routes that return a single object but not for e.g. `RHCheckinAPIRegForms` nor `RHCheckinAPIRegistrations` which also just get the details

Reply:
My idea was that the `...Details` endpoints are more detailed than just the lists. For example, the regform/registration details include the actual list of registrations / registration data, while the endpoints that just return the list of all regforms/registrations don't.